plugins/permission-common/report.api.md

```diff
@@ -183,9 +183,19 @@ export class PermissionClient implements PermissionEvaluator {
   ): Promise<QueryPermissionResponse[]>;
 }
 
+// @public
+export type PermissionClientOptions = {
+  discovery: DiscoveryApi;
+  config: Config;
+  loaderCacheTtl?: number;
+  cacheTtl?: number;
+  batchDelay?: number;
+};
+
 // @public
 export type PermissionClientRequestOptions = {
   token?: string;
+  identifier?: string;
```

Comment:
A fairly generic name - I don't think it has a doc comment either right? Naming is hard.

Especially in permission-node we could actually make it a credentials object instead, making it an internal concern what the client does with it at least from the backend standpoint, right? And a similar thing on the frontend side, exposing the backend identity (the return value from the identity service), or even removing the parameter entirely and forcibly getting the identity inside the implementation instead. Compare the distinction between the catalogService api on the node side being different from the frontend catalog utility api.

Reply:
Yes, the issue is that there is no common type that works for both backend and frontend (AFAIK). In the frontend, you can get the `BackstageUserIdentity` from `IdentityApi`, and in the backend, you can get `BackstageCredentials`.  There's not really a common ground between these two, and it's not possible to import the frontend type in the `PermissionService`.

Reply:
I created a new `PermissionIdentity` type that forces the identifier to be in specific format. The stringifying is then done inside the `PermissionClient` internally.